sympy/integrals/tests/test_integrals.py

```diff
@@ -1893,6 +1893,16 @@ def test_issue_21024():
     assert F_true == integrate(f, x)
 
 
+def test_issue_21721():
+    a = Symbol('a')
+    b = Symbol('b', real=True)
+    expr = -I*(-2*I*pi*Heaviside(im(a) - 1, 0) + 2*I*pi*Heaviside(im(a) + 1, 0))/(2*pi)
+    assert integrate(1/(pi*(1+(x-a)**2)),(x,-oo,oo)) == expr
+    assert integrate(1/(pi*(1+(x-b)**2)),(x,-oo,oo)) == 1
+    for a in [-I/2, I/2]:
+        assert integrate(1/(pi*(1+(x-a)**2)),(x,-oo,oo)) == 1
```

Comment:
These tests are not needed when the antiderivative is given above.

Reply:
Sure , will change them . It's just that all other cases were working on master except this one which is `-1< im(a) < 1` . This case used to give `0` on master, so what we really addressed here was this case. But we anyways have the integral in the form of parameter `a`, so we can get rid of this .